service/web/publisherserver.go

```diff
@@ -34,11 +35,10 @@ func (s publisherServer) Publish(ctx context.Context, req *metrov1.PublishReques
 		"topic": req.Topic,
 	})
 	defer span.Finish()
-
-	if ok, err := s.topicCore.ExistsWithName(ctx, req.Topic); err != nil {
-		return nil, merror.ToGRPCError(err)
-	} else if !ok {
-		return nil, merror.New(merror.NotFound, "topic not found").ToGRPCError()
+	if tasks.CheckIfTopicExists(ctx, req.Topic) {
+		logger.Ctx(ctx).Infow("Topic exists inside the cache..", "req", req.Topic)
+	} else {
```

Comment:
In case of topic missing from the cache, we need to fall back to the original consul verify flow.
Caching helps us improve the ingestion latency for cases where the topic exists in the cache, but the cache should not be the source of truth here.